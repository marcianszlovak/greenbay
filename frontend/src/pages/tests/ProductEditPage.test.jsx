import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import ProductEditPage from '../ProductEditPage';
import store from '../../store';

describe('ProductEditPage Component', () => {
  const MockProductEditPage = (
    <MemoryRouter>
      <Provider store={store}>
        <ProductEditPage match={{ params: { id: 1 } }} />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockProductEditPage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
