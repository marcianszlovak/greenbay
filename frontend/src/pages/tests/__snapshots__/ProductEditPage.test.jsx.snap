// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductEditPage Component should match the snapshot 1`] = `
Array [
  <a
    className="btn btn-light my-3"
    href="/admin/productlist"
    onClick={[Function]}
  >
    Go Back
  </a>,
  <div
    className="container"
  >
    <div
      className="justify-content-md-center row"
    >
      <div
        className="col-md-6 col-12"
      >
        <h1>
          Edit Product
        </h1>
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              id="name"
              onChange={[Function]}
              placeholder="Enter name"
              type="name"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="price"
            >
              Price
            </label>
            <input
              className="form-control"
              id="price"
              onChange={[Function]}
              placeholder="Enter price"
              type="number"
              value={0}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="image"
            >
              Image
            </label>
            <input
              className="form-control"
              id="image"
              onChange={[Function]}
              placeholder="Enter image url"
              type="text"
              value=""
            />
            <div
              className="custom custom-file"
            >
              <input
                className="custom-file-input"
                disabled={false}
                id="image-file"
                onChange={[Function]}
                type="file"
              />
              <label
                className="custom-file-label"
                htmlFor="image-file"
              >
                Choose File
              </label>
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="brand"
            >
              Brand
            </label>
            <input
              className="form-control"
              id="brand"
              onChange={[Function]}
              placeholder="Enter brand"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="countInStock"
            >
              Count In Stock
            </label>
            <input
              className="form-control"
              id="countInStock"
              onChange={[Function]}
              placeholder="Enter countInStock"
              type="number"
              value={0}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="category"
            >
              Category
            </label>
            <input
              className="form-control"
              id="category"
              onChange={[Function]}
              placeholder="Enter category"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="description"
            >
              Description
            </label>
            <input
              className="form-control"
              id="description"
              onChange={[Function]}
              placeholder="Enter description"
              type="text"
              value=""
            />
          </div>
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            Update
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
